base/replutil.jl

```diff
@@ -156,6 +156,13 @@ function showerror(io::IO, e::MethodError)
         print(io, "\nNote the difference between 1d column vector [1,2,3] and 2d row vector [1 2 3].")
         print(io, "\nYou can convert to a column vector with the vec() function.")
     end
+    # Give a helpful error message if the user likely called a type constructor
+    # and sees a no method error for convert
+    if e.f == Base.convert && !isempty(e.args) && isa(e.args[1], Type)
+        println(io)
+        print(io, "This may have arisen from a call to the constructor $(e.args[1])(...), ")
+        print(io, "since type constructors fall back to convert methods in julia v0.4.")
```

Comment:
I'd avoid mentioning a specific version number here, this runs the risk of going stale if people forget about it.